src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java

```diff
@@ -64,7 +65,7 @@ public void secretFileBaseTestWithDeprecatedCtor() throws IOException, URISyntax
     }
 
     private void secretFileTest(boolean useDeprecatedConstructor) throws IOException, URISyntaxException {
-        DiskFileItem fileItem = createEmptyFileItem();
+        FileItem fileItem = createEmptyFileItem();
```

Comment:
> also relying on a Jenkins API rather than a particular third-party implementation will allow us to swap out the implementation in the future in https://github.com/jenkinsci/jenkins/pull/9259 without breaking this test.

this is `org.apache.commons.fileupload.FileItem` so does not acheive this part of the intended by this PR.

Reply:
Yes exactly; your understanding is correct. I'll clarify this in detail when I write this up.